src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverFactoryConfig.java

```diff
@@ -149,6 +149,12 @@
                      " and on the alias update time if the number of aliases is huge (over 10000).")
     boolean resource_resolver_optimize_alias_resolution() default true;
 
+    @AttributeDefinition(name = "Allowed Optimize alias path",
+         description = "This setting can contain a list of path prefixes, e.g. /libs/, /content/. If " +
+                 "such a list is configured, for alias optimization, only paths from resources starting with this prefix " +
+                 "are considered. If the list is empty, all paths are used.)")
+    String[] resource_resolver_optimize_alias_allowedlist();
```

Comment:
Nit: I think "allowlist" is cleared than "allowedlist"

Reply:
The equivalent property for vanity paths is called `resource_resolver_vanitypath_whitelist`. For consistency's sake I think this should be called `resource_resolver_alias_whitelist` or maybe slightly more descriptive `resource_resolver_alias_location_whitelist`.

Reply:
I would say that since we started discussing about https://cwiki.apache.org/confluence/display/SLING/Removal+of+problematic+language we should not add any more problematic terms like whitelist.

Reply:
Agreed. We can use unproblematic language and communicate the intent more clearly at the same time. What about `resource_resolver_allowed_alias_locations`?

Reply:
That would work for me.